target/linux/layerscape/image/mkits-multiple-config.sh

```diff
@@ -0,0 +1,594 @@
+			signature@${count} {
+				algo = "${algo}";
+				key-name-hint = "${key}";
+EOF
+			if [ -n "$SIGN_IN_CONF" ] ; then
+				echo "			sign-images = \"$img\";" >> ${OUTPUT}
+			fi
+			echo "			};" >> ${OUTPUT}
+
+			count=`expr ${count} + 1`
+		done
+	done
+}
+
+#
+# Emit config sub nodes
+#
+emit_config() {
+	local conf_csum="sha1"
+
+	config_name="conf@${1}"
+	if [ ! -z "${11}" ]; then
+		config_name="${11}"
+	fi 
+	if [ -z "${2}" ]; then
+		echo "Error: config has no kernel img, skipping conf node!"
+		return 0
+	fi
+
+	# Test if we have any DTBs at all
+	if [ -z "${3}" ] ; then
+		conf_desc="Boot Linux kernel"
+		fdt_line=""
+	else
+		conf_desc="Boot Linux kernel with FDT blob"
+		fdt_line="
+			fdt = \"fdt@${3}\";"
+	fi
+
+	# Test if we have any ROOTFS at all
+	if [ -n "${4}" ] ; then
+		conf_desc="$conf_desc + ramdisk"
+		fdt_line="${fdt_line}
+			ramdisk = \"ramdisk@${4}\";"
+	fi
+
+	kernel_line="kernel = \"kernel@${2}\";"
+
+	cat << EOF >> ${OUTPUT}
+		${config_name} {
+			description = "${conf_desc}";
+			${kernel_line}${fdt_line}
+			hash@1 {
+				algo = "${conf_csum}";
+			};
+EOF
+	if [ -n "$SIGN_IN_CONF" ] ; then
+		emit_signature "$5" "$6" "$7" "$8" "$9" "${10}"
+	fi
+
+	echo "		};" >> ${OUTPUT}
+}
+
+#
+# remove prefix space
+#
+remove_prefix_space()
+{
+	echo "$@" | sed "s:^ ::g"
+}
+
+#
+# generate image nodes and its subnodes
+#
+emit_image_nodes()
+{
+	local t img_c img_i img_index chk
+	local img_type img_path img_count img_loadadr img_entrypoint \
+		img_compression img_hash img_conf img_name img_key img_sign
+
+	emit_its imagestart
+	for t in "kernel" "fdt" "ramdisk"; do
+		img_index=0
+		for a in ${img_array[@]}; do
+			img_type=$(array_get $a 0)
+			img_path=$(array_get $a 1)
+			img_count=$(array_get $a 2)
+			img_loadadr=$(array_get $a 3)
+			img_entrypoint=$(array_get $a 4)
+			img_compression=$(array_get $a 5)
+			img_hash=$(array_get $a 6)
+			img_conf=$(array_get $a 7)
+			img_name=$(array_get $a 8)
+			img_key=$(array_get $a 9)
+			img_sign=$(array_get $a 10)
+			img_cname=$(array_get $a 11)
+			
+			img_conf=$(remove_prefix_space $img_conf)
+			img_hash=$(remove_prefix_space $img_hash)
+
+			[ "${img_type}" == $t ] || continue
+			# generate sub nodes
+			eval chk=\$DEF_$t
+			[ -n "${chk}" ] || eval DEF_$t=$img_count
+			case $t in
+				kernel) emit_kernel "$img_path" "$img_count" \
+					"$img_loadadr" "$img_entrypoint" \
+					"$img_compression" "$img_hash" \
+					"$img_name" "$img_key" "$img_sign";;
+				fdt) emit_fdt "$img_path" "$img_count" \
+					"$img_compression" "$img_hash" \
+					"$img_name" "$img_loadadr" "$img_key" "$img_sign"  ;;
+
+				ramdisk) emit_ramdisk "$img_path" "$img_count" \
+					"$img_compression" "$img_hash" \
+					"$img_name" "$img_key" "$img_sign";;
+			esac
+
+			# set up configuration data
+			for img_c in ${img_conf}; do
+				img_i=""
+				#set up default configuration if its not set
+				[ -n "$DEF_CONFIG" ] || DEF_CONFIG=$img_c
+				[ -n "${img_c}" ] || continue
```

Comment:
You've already tested for non-empty here so the next for empty on the next line is redundant.